### PR TITLE
Initial Import of ServerManagement SDK

### PR DESCRIPTION
**Pull request by fearthecowboy**
(into AutoRest branch this time!)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@fearthecowboy__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (garretts). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/Extensions.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
+using System;
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/Extensions.cs]: @fearthecowboy please add license header in all files

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("KeyVault.Management.Tests")]
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs]: KeyValt assembly title?

File: src/ResourceManagement/ServerManagement/ServerManagement/Generated/GatewayOperationsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,738 @@
+
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement/Generated/GatewayOperationsExtensions.cs]: AutoRest should generate headers by default. Please regen it after you wil lremove -Header NONE option in generate.cmd

File: src/ResourceManagement/ServerManagement/ServerManagement/generate.cmd [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+:: Microsoft Azure SDK for Net - Generate library code
+:: Copyright (C) Microsoft Corporation. All Rights Reserved.
+::
+ 
+set autoRestVersion=0.16.0-Nightly20160329
+set specUrl="https://raw.githubusercontent.com/fearthecowboy/azure-rest-api-specs/master/arm-servermanagement/2015-07-01-preview/servermanagement.json"
+set source=-Source https://www.myget.org/F/autorest/api/v2
+
+set repoRoot=%~dp0..\..\..\..
+set autoRestExe=%repoRoot%\packages\autorest.%autoRestVersion%\tools\AutoRest.exe
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement/generate.cmd]: Any reason why you opt out from using default generator.cmd? e.g. https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/src/ResourceManagement/Compute/Microsoft.Azure.Management.Compute/generate.cmd

**Comment by hovsepm** [comment]
@fearthecowboy please convert your project to DNX. All our autorest based SDKs should be in a DNX projects. We will convert them later to CoreClr

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (fearthecowboy) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs]: Borrowed file. Not that a assembly title in a test project was all that valuable :)

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/Extensions.cs (hunk 1 above)
Comment (fearthecowboy) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/Extensions.cs]: And that's what I got for borrowing the wrong generate script.

File: src/ResourceManagement/ServerManagement/ServerManagement/generate.cmd (hunk 4 above)
Comment (fearthecowboy) [inline src/ResourceManagement/ServerManagement/ServerManagement/generate.cmd]: I just borrowed the code from another project in the repo.

I guess I borrowed the wrong one. *sigh*

**Comment by fearthecowboy** [comment]
@hovsepm ... DNX is dead. long live dotnet-cli :)

Hmm. I don't suppose I can get to that a bit later?  

Seriously, I recently had to wipe out the DNX bits of my box to correct a conflict between that and work I was doing with dotnet-cli.

**Comment by hovsepm** [comment]
@fearthecowboy this is the PR that fails on DNX workers but works on Core CLR workers - https://github.com/Azure/azure-sdk-for-net/pull/1939 and we are waiting until CLR will go public to merge netcore branch to AutoRest. You will have to do migration yourself some time in the very near future (probably by the end of this month) if you will not convert your project now. 
Plus I'm not sure you will be able to publish your package form AutoRest branch to nuget if it is not DNX. That is a path of trial and possible failures.

File: src/ResourceManagement/ServerManagement/ServerManagement/ClientMessageType.cs [hunk 5]
```diff
@@ -0,0 +1,151 @@
+﻿using System;
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement/ClientMessageType.cs]: Header missing in this file.

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/ServerManagementTestBase.cs [hunk 6]
```diff
@@ -0,0 +1,352 @@
+﻿using System;
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/ServerManagementTestBase.cs]: Header missing in this file.

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/Properties/AssemblyInfo.cs]: Header missing in this file.

File: src/ResourceManagement/ServerManagement/ServerManagement.Tests/OperationsTest.cs [hunk 8]
```diff
@@ -0,0 +1,429 @@
+﻿using System;
```
Comment (hovsepm) [inline src/ResourceManagement/ServerManagement/ServerManagement.Tests/OperationsTest.cs]: Header missing in this file.

**Comment by fearthecowboy** [comment]
Fixed

**Pull request merged**